service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java

```diff
@@ -64,7 +64,11 @@ public InputBundle(Bundle bundle) {
         var resourceType = ResourceType.fromCode(reference.getResourceType());
 
         if (MAPPABLE_RESOURCES.contains(resourceType)) {
-            return extractResourceByReference(this.bundle, reference);
+            Optional<Resource> resource = extractResourceByReference(this.bundle, reference);
+            if (resource.isPresent()) {
+                return resource;
+            }
+            throw new EhrMapperException("Resource not found: " + reference);
```

Comment:
Are we sure this should be changed from Reference?

Reply:
Following conversation with @dmcavenue1, 'Resource not found' is acceptable